settings.gradle

```diff
@@ -103,6 +104,15 @@ if (localBuilds.exists()) {
             }
         }
     }
+
+    if (ext.has("localAboutAutomatticPath")) {
+        includeBuild(ext.localAboutAutomatticPath) {
+            dependencySubstitution {
+                println "Substituting about-automatiic with the local build"
+                substitute module("$gradle.ext.aboutAutomatticBinaryPath") with project(':AboutAutomattic')
```

Comment:
When I use a local build, this fails:

```
Substituting about-automatiic with the local build

FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/gio/Developer/a8c/wpandroid/settings.gradle' line: 112

* What went wrong:
Project :AboutAutomattic not found.
```

This is the `local-builds.gradle` I'm using

```
ext {
    localAboutAutomatticPath = "../about-automattic-android"
    //localFluxCPath = "../WordPress-FluxC-Android"
    //localWPUtilsPath = "../WordPress-Utils-Android"
    //localGutenbergMobilePath = "../gutenberg-mobile"
    //localLoginFlowPath = "../WordPress-Login-Flow-Android"
    //localStoriesAndroidPath = "../stories-android"
    //localAztecAndroidPath = "../AztecEditor-Android"
}
```

And I verified the path is correct 😄 : 

<img alt="image" src="https://user-images.githubusercontent.com/1218433/142559790-35bf0865-2a98-4816-8ce1-fb0ac231bb83.png" width=500/>

Reply:
I found the issue with the above.  Looks like it was renamed to `library` in `automattic-android-about`  repo hence it is not finding it.  I changed to :library at line 112 in settings.gradle that resolves it. 

However, that's not the reason for lint failure though

Reply:
The issue could be with this

`Could not load custom lint check jar file .gradle/caches/transforms-3/03a4dea5bd244b5ca160f0ea25e79cb3/transformed/jetified-animation-core-1.1.0-beta02/jars/lint.jar`

We've just started using Jepack Compose (a.k.a android SwiftUI).  Lint seems to have problem with it.  I'm not sure why it is not a problem on the about-automattic-android library where it is used!  Also, on WPAndroid detekt and ktlint seems to pass fine!

Reply:
Looking at the lint results document, it looks like there are several failures going on. One of them produces the following error message:

```
  | [2021-11-19T05:06:42Z]   /workdir/WordPress/src/main/java/org/wordpress/android/push/NativeNotificationsUtils.java:70: Error: Missing permissions required by intent Intent.ACTION_CLOSE_SYSTEM_DIALOGS: android.permission.BROADCAST_CLOSE_SYSTEM_DIALOGS [MissingPermission]
  | [2021-11-19T05:06:42Z]           context.sendBroadcast(closeIntent);
  | [2021-11-19T05:06:42Z]                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  | [2021-11-19T05:06:42Z]   /workdir/WordPress/src/main/java/org/wordpress/android/ui/notifications/ShareAndDismissNotificationReceiver.java:21: Error: Missing permissions required by intent Intent.ACTION_CLOSE_SYSTEM_DIALOGS: android.permission.BROADCAST_CLOSE_SYSTEM_DIALOGS [MissingPermission]
  | [2021-11-19T05:06:42Z]           context.sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
  | [2021-11-19T05:06:42Z]                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This seems to indicate that this particular failure occurs because we're missing the permission to close system dialogs, [which is a requirement for Android 12](https://developer.android.com/about/versions/12/behavior-changes-all#close-system-dialogs). According to that page, this change affects **all apps**: 

>The following behavior changes apply to _all apps_ when they run on Android 12, regardless of `targetSdkVersion`. You should test your app and then modify it as needed to support these properly, where applicable.

If I had to guess, I'd say these issues are now visible because we bumped `compileSdkVersion` to `31`. Maybe we should handle those changes on a separate PR before merging this.

Let me know what you think.

Reply:
I thought so too at first.  Then I tested it on develop branch without that change and it's happening on develop too!